agent/stats/queue.go

```diff
@@ -137,6 +138,101 @@ func (queue *Queue) GetStorageWriteStatsSet() (*ecstcs.ULongStatsSet, error) {
 	return queue.getULongStatsSet(getStorageWriteBytes)
 }
 
+// GetNetworkStatsSet gets the stats set for network metrics.
+func (queue *Queue) GetNetworkStatsSet() (*ecstcs.NetworkStatsSet, error) {
```

Comment:
do we need to lock the queue for this and like methods?  
I see that the `GetRawUsageStats()` Queue method locks during its execution

Reply:
this one ultimately calls `getULongStatsSet` , which has lock on the queue.